mod/state-transition/pkg/core/state_processor_staking.go

```
@@ -248,76 +248,74 @@ func (sp *StateProcessor[
+		return err
 	}
 
 	totalValidators, err := st.GetTotalValidators()
 	if err != nil {
 		return err
 	}
 
-	// Update the next validator index to start the next withdrawal sweep
+	// Update the next validator index to start the next withdrawal sweep.
 	//#nosec:G701 // won't overflow in practice.
 	if numWithdrawals == int(sp.cs.MaxWithdrawalsPerPayload()) {
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion uint64 -> int

(gosec)